### PR TITLE
doc: it8xxx2: Documentation improvements

### PR DESCRIPTION
**Pull request by fuhaolei**
Better documentation describing the flashing and booting process.

Signed-off-by: Fu Haolei <haolei.fu@intel.com>

### CODE REVIEW

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 1]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+     - adc
+   * - I2C
+     - on-chip
+     - i2c
+   * - KSCAN
+     - on-chip
+     - kscan
+
+
+Other hardware features are not currently supported by Zephyr (at the moment)
```
Comment (keith-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:73]: Nit - no need to include "(at the moment)".  Please add a closing period "."

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 2]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+
+The default configuration can be found in the
+:zephyr_file:`boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig` Kconfig file.
+
+Programming and debugging on it83202
+************************************
+
```
Comment (keith-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:80]: Move the requirements to the front of this section:
Download Board
Flash tool
it8xxx2_evb board

Please also provide links for where the user can obtain these.

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 3]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+Wiring
+=======
+#. Connect your DownLoad Board to your host computer using the USB cable.
```
Comment (keith-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:83]: Please use a consistent name  for Download Board - you have the following variations:
"DownLoad Board"
"Download Board"
"download board"

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 4]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+#. Build :ref:`hello_world` application as you would normally do
+   (see :`Zephyr Getting Started Guide`_):.
 
 .. zephyr-app-commands::
    :board: it8xxx2_evb
+   :zephyr-app: samples/hello_world
    :goals: build
 
+#. The file :file:`zephyr.bin` will be created if the build system
+   can find the image generation tool.
```
Comment (keith-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:144]: Is the image generation tool separate from Zephyr?  Please provide instructions to get this tool.

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 5]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+--------
+
 You can get the winflash tool at here:`Download Link`_
 Use the winflash.exe tool to program a zephyr application
 to the it8xxx2 board flash.
 
+#. Open winflash tool and make sure the order you open the switch of DownLoad
+   Board is right. Configure your flash tool like below.
+
+    .. figure:: WinFlashTool_P2.png
```
Comment (keith-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:162]: WinFlashTool_P2.png is a poor quality image.  Can you provide a better or higher resolution image?

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 6]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+   First, click ``Load`` button and select your zephyr.bin file.
+   Second, click ``run`` to flash the iamge into board.
+
+    .. figure:: WinFlashTool_P3.png
+          :width: 600px
+          :align: center
+
+#. Until here, you have flashed your image into ITE board and
+   it will work if you turn on ITE board. We can use MobaXterm
+   to test result.
```
Comment (keith-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:180]: Make this text match the Ubuntu instructions.  "Use a terminal program to verify flashing worked correctly. You should see "Hello World! it8xxx2_evb" sent by the board.  If you don't see this message, press the Reset button and the message should appear."

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 7]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
 Debugging
 =========
 
 Supporting uart debug, currently.
 
+Troubleshooting
+===============
+
+#. If your flash tool report a fail, re-plug the 8390 download board or
+   power on the ec and try again.
```
Comment (keith-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:213]: "If the flash tool reports a failure..."

**Review by keith-zephyr — DISMISSED** [review]

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 8]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+
+Windows
+--------
+
 You can get the winflash tool at here:`Download Link`_
 Use the winflash.exe tool to program a zephyr application
 to the it8xxx2 board flash.
 
+#. Open winflash tool and make sure the order you open the switch of DownLoad
+   Board is right. Configure your flash tool like below.
```
Comment (brockus-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:160]: How do you know the order the switch is opened is "right"?  Being concrete here may make it easier for users that want to use this tool

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 9]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+
+#. Using winflash tool flash zephyr.bin into your ITE board.
+   First, click ``Load`` button and select your zephyr.bin file.
+   Second, click ``run`` to flash the iamge into board.
+
+    .. figure:: WinFlashTool_P3.png
+          :width: 600px
+          :align: center
+
+#. Until here, you have flashed your image into ITE board and
```
Comment (brockus-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:178]: "Until here" sounds awkward to me. I would make this "At this point"

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 10]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+#. Run your favorite terminal program to listen for output.
+   Under Linux the terminal should be :code:`/dev/ttyUSB0`. Do not close it.
+
+   For example:
+
+   .. code-block:: console
+
+      $ minicom -D /dev/ttyUSB0 -b 115200
+
+#. Flash your board using ``west`` from the second terminal window.
```
Comment (brockus-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:194]: Did I miss this information on opening a second terminal window?  I only see the instructions to open /dev/ttyUSB0.

File: boards/riscv/it8xxx2_evb/doc/index.rst (hunk 1 above)
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: Sorry for delay. Fixed it.

File: boards/riscv/it8xxx2_evb/doc/index.rst (hunk 3 above)
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: Fixed.

File: boards/riscv/it8xxx2_evb/doc/index.rst (hunk 4 above)
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: The image generation tool is west, modified.

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 11]
```diff
@@ -60,34 +59,159 @@ currently supports the following hardware features:
...
+you'll need our flash tool and download board.
+
+Windows
+--------
+
 You can get the winflash tool at here:`Download Link`_
 Use the winflash.exe tool to program a zephyr application
 to the it8xxx2 board flash.
 
+#. Open winflash tool and make sure the order you open the switch of DownLoad
```
[the remark this answers is not in the document]
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: Fixed.

File: boards/riscv/it8xxx2_evb/doc/index.rst (hunk 8 above)
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: Sorry for delay. Provided more details.

File: boards/riscv/it8xxx2_evb/doc/index.rst (hunk 9 above)
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: Ah..Fixed.

File: boards/riscv/it8xxx2_evb/doc/index.rst (hunk 10 above)
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: Just use west on the second terminal window, modified.

File: boards/riscv/it8xxx2_evb/doc/index.rst (hunk 2 above)
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst:80]: Added one link here.

File: boards/riscv/it8xxx2_evb/doc/index.rst (hunk 6 above)
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: This section is about flashing on windows, updated.

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 12]
```diff
@@ -60,34 +59,171 @@ currently supports the following hardware features:
...
 Debugging
 =========
 
 Supporting uart debug, currently.
 
+Troubleshooting
+===============
+
+#. If the flash tool reports a failure, re-plug the 8390 Download board or
+   power on the ec and try again.
```
Comment (keith-zephyr) [inline boards/riscv/it8xxx2_evb/doc/index.rst:225]: "power on the ec"?  Do you mean power cycle the it8xxx2_evb board?

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 13]
```diff
@@ -60,34 +59,171 @@ currently supports the following hardware features:
...
+:zephyr_file:`boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig` Kconfig file.
+
+Programming and debugging on it83202
+************************************
 
-Programming and debugging
-*************************
+In order to upload the application to the device,
+you'll need our flash tool and Download board.
+You can get them at:`ITE's website`_.
```
Comment (jackrosenthal) [inline boards/riscv/it8xxx2_evb/doc/index.rst:83]: This is only required for Windows users, right?

If so, add that...

> This tool is only required for Windows users. For Linux users, `west flash` can flash the board without additional tooling.

Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst:83]: Yes. And maybe I need to find a place to store the linux version flash tool.

File: boards/riscv/it8xxx2_evb/doc/index.rst (hunk 12 above)
Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: Yes, "ec" is confusing. I have fixed it.

**Comment by fuhaolei** [comment]
The linux flash tool has been uploaded to ITE's website.

**Comment by fuhaolei** [comment]
Hi，@keith-zephyr, @jackrosenthal , @brockus-zephyr , I have addressed your comments, could you help me review my PR?

**Review by keith-zephyr — DISMISSED** [review]

**Review by chen-png — DISMISSED** [review]

File: boards/riscv/it8xxx2_evb/doc/index.rst [hunk 14]
```diff
@@ -60,39 +59,172 @@ currently supports the following hardware features:
...
+
+The default configuration can be found in the
+:zephyr_file:`boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig` Kconfig file.
+
+Programming and debugging on it83202
+************************************
+
+In order to upload the application to the device,
+you'll need our flash tool and Download board.
+You can get them at:`ITE's website`_.
```
Comment (edersondisouza) [inline boards/riscv/it8xxx2_evb/doc/index.rst:83]: Maybe add a space between `:` and the link? Seeing them joined together makes me feel annoyed, but maybe it's just me. This also happens in the previous reference to ITE website in the doc (which wasn't touched in this PR).

Comment (fuhaolei) [reply boards/riscv/it8xxx2_evb/doc/index.rst]: Sorry, it's my fault. Fixed.

**Review by edersondisouza — APPROVED** [review]

**Review by chen-png — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**